### PR TITLE
PXP-538: Renew auth token

### PR DESCRIPTION
**Pull request by mfauzaan**
## Summary
This pull request aims to enhance our Vault integration by implementing auth token renewal using the Vault API's auto-renew endpoint. Currently, wukong cli requires users to manually re-login to Vault after their tokens expire, which can be inconvenient for users making multiple Vault API calls.

Auto-Renew Endpoint Documentation: [Renew a Token (Self)](https://developer.hashicorp.com/vault/api-docs/auth/token#renew-a-token-self)

Ticket: [PXP-538](https://mindvalley.atlassian.net/browse/PXP-538)

## What's Changed
- [x] Add renewal endpoint to client vault
- [x] Utilized the auto-renew endpoint to extend the expiration time of tokens.
- [x] Renew the token if expiry is less then an hour

[PXP-538]: https://mindvalley.atlassian.net/browse/PXP-538?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/103#" title="Author self-approved">mfauzaan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: src/config.rs
Comment (mfauzaan) [reply src/config.rs:71]: This is to support the backward compatibility, as some users already has the token without expiry

Is there any workaround for this?

Comment (jk-gan) [reply src/config.rs:71]: I see.

@onimsha since this is still pre 1.0, can we still introduce breaking change? Users have to run `wukong init` again to make it work.

Comment (onimsha) [reply src/config.rs:71]: > I see.
> 
> @onimsha since this is still pre 1.0, can we still introduce breaking change? Users have to run `wukong init` again to make it work.

Yes, I guess we can ask them to do it one more time.

File: src/error.rs
Comment (onimsha) [inline src/error.rs:200]: @mfauzaan @jk-gan Running `wukong init` won't have anything to do with the `.wukong.toml` right ? Wanna make sure about what we're should tell our user here.

Comment (mfauzaan) [reply src/error.rs:200]: So the behaviour i know is
When we do `wukong init` and if there is mismatch of the config, it resets to the config file to default state basically clearing out the existing data

Ref: https://github.com/mindvalley/wukong-cli/blob/main/src/commands/init.rs#L23

Comment (onimsha) [reply src/error.rs:200]: ah sorry I was confusing this with the `.wukong.toml` config. You can disregard this comment.

**Review by onimsha — APPROVED** [review]

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/103#" title="Author self-approved">mfauzaan</a>*, *<a href="https://github.com/mindvalley/wukong-cli/pull/103#pullrequestreview-1471342176" title="Approved">onimsha</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
